ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -41,27 +40,32 @@ object EndpointsCompanion {
 
   final case class Unauthorized(message: String) extends Error
 
-  final case class ServerError(message: String) extends Error
+  final case class ServerError(message: Throwable \/ String) extends Error
 
   final case class ParticipantServerError(grpcStatus: GrpcCode, description: Option[String])
       extends Error
 
   final case class NotFound(message: String) extends Error
 
+  object ServerError {
+    def apply(message: String): ServerError = ServerError(\/.right(message))
+    def apply(error: Throwable): ServerError = ServerError(\/.left(error))
+  }
+
   object Error {
     implicit val ShowInstance: Show[Error] = Show shows {
       case InvalidUserInput(e) => s"Endpoints.InvalidUserInput: ${e: String}"
       case ParticipantServerError(s, d) =>
         s"Endpoints.ParticipantServerError: ${s: GrpcCode}${d.cata((": " + _), "")}"
-      case ServerError(e) => s"Endpoints.ServerError: ${e: String}"
+      case ServerError(e) => s"Endpoints.ServerError: ${e.fold(_.getMessage, identity): String}"
```

Comment:
Since you always get the message out here, I don't think you need to track it in a disjunction